content/blog/2021-11-17-exp-versioning.md

```diff
@@ -0,0 +1,243 @@
+  plots:
+  - logs.csv:
+      cache: false
+Do you want to add the above contents to dvc.yaml? [y/n]: y
+
+Created default stage in dvc.yaml. To run, use "dvc exp run".
+See https://dvc.org/doc/user-guide/experiment-management/running-experiments.
+```
+
+DVC treats experiments as code. Rather than log parameters and metrics with API
+calls, DVC expects them to be saved in files. This might mean a little more work
+to set up your code to read parameters from a YAML file or write metrics to a
+JSON file. However, once you set up your repo in this structure, you start to
+see the benefits of DVC:
+
+- Experiment information lives in readable files that are always available even
+  without DVC, and your code stays clean.
+- Your code and experiments stay connected in the repo, and you can take
+  advantage of all of the features of Git, like diffing and branching.
+- Your entire experiment is codified. Notice that the command above generated a
+  `dvc.yaml` file that tracks the experiment pipeline and all its inputs and
+  outputs.
+
+Of course, you could do all this yourself in Git. Since we have established that
+code version control is insufficient for experiments, DVC adds features specific
+to ML needs.
+
+Large data and models are tracked in your own storage yet are Git-compatible.
+For example, the above command tracks the `models` folder in DVC, making Git
+ignore it yet storing and versioning it for you so that you can back up versions
+anywhere and check them out alongside your experiment code.
+
+Experiments are tracked by DVC so you don't need to create commits or branches
+for each one, yet they are also Git compatible and can be converted into Git
+branches or merged into your existing ones.
+
+```dvc
+$ dvc exp show
+┏━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━┳━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━┓
+┃ Experiment              ┃ Created      ┃    loss ┃    acc ┃ train.epochs ┃
+model.conv_units ┃
+┡━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━╇━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━┩
+│ workspace               │ -            │ 0.25183 │ 0.9137 │ 10           │ 64
+│
+│ dvc-init                │ Oct 23, 2021 │       - │      - │ 10           │ 16
+│
+│ ├── 9a4ff1c [exp-333c9] │ 10:40 AM     │ 0.25183 │ 0.9137 │ 10           │ 64
+│
+│ ├── 138e6ea [exp-55e90] │ 10:28 AM     │ 0.25784 │ 0.9084 │ 10           │ 32
+│
+│ └── 51b0324 [exp-2b728] │ 10:17 AM     │ 0.25829 │ 0.9058 │ 10           │ 16
+│
+└─────────────────────────┴──────────────┴─────────┴────────┴──────────────┴──────────────────┘
+
+$ dvc exp branch exp-333c9 conv-units-64
+Git branch 'conv-units-64' has been created from experiment 'exp-333c9'.
+To switch to the new branch run:
+
+        git checkout conv-units-64
+
+```
+
+Reproducing experiments can be automated since your experiment pipeline is
+codified, and DVC will check for cached inputs and outputs and skip recomputing
+data that's been generated before.
+
+```dvc
+$ dvc exp run
```

Comment:
Value of this step might be not clear. Should user promote this exp or continue working on this idea - run next exp?